addons/mail/static/tests/chat_window/chat_window_tests.js

```diff
@@ -1244,3 +1245,42 @@ QUnit.test(
         assert.containsOnce($, ".o-mail-ChatWindow");
     }
 );
+
+QUnit.test("hidding chat window does not fold it", async (assert) => {
+    patchUiSize({ size: SIZES.MD }); // only 2 chat window can be opened at a time
+    const pyEnv = await startServer();
+    pyEnv["discuss.channel"].create([{ name: "General" }, { name: "Sales" }, { name: "D&D" }]);
+    await start({
+        mockRPC(route, args) {
+            if (args.method === "channel_fold") {
+                const [channel] = pyEnv["discuss.channel"].searchRead([
+                    ["id", "=", args.args[0][0]],
+                ]);
+                assert.step(`${channel.name} - ${args.kwargs.state}`);
+            }
+        },
+    });
+    await clickContains(".o_menu_systray i[aria-label='Messages']");
+    await clickContains(".o-mail-NotificationItem", { text: "General" });
+    await contains(".o-mail-ChatWindow", { text: "General" });
+    assert.verifySteps(["General - open"]);
+    await clickContains(".o_menu_systray i[aria-label='Messages']");
+    await clickContains(".o-mail-NotificationItem", { text: "Sales" });
+    await contains(".o-mail-ChatWindow", { text: "Sales" });
+    assert.verifySteps(["Sales - open"]);
+    // Sales chat window will be hidden since there is not enough space for the
+    // D&D one.
+    await clickContains(".o_menu_systray i[aria-label='Messages']");
+    await clickContains(".o-mail-NotificationItem", { text: "D&D" });
+    assert.verifySteps(["D&D - open"]);
+    await contains(".o-mail-ChatWindow", { text: "D&D" });
+    // D&D chat window will be hidden since there is not enough space for the
+    // Sales one.
+    await clickContains(".o-mail-ChatWindowHiddenToggler");
+    await clickContains(".o-mail-ChatWindowHiddenMenu-item .o-mail-ChatWindow-name", {
+        text: "Sales",
+        visible: true,
+    });
+    await contains(".o-mail-ChatWindow", { text: "Sales" });
+    assert.verifySteps(["Sales - open"]);
```

Comment:
This doesn't look correct, in terms of the server "sales" should already have been open at 1270, the hidden menu should be irrelevant to that and the rcp shouldn't be made again.